builder/azure/arm/builder.go

```diff
@@ -91,6 +92,12 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 		return nil, err
 	}
 
+	b.config.ObjectID = getObjectIdFromToken(spnCloud)
+
+	if b.config.ObjectID == "" && b.config.OSType != constants.Target_Linux {
+		return nil, fmt.Errorf("could not determined the ObjectID for the user, which is required for Windows builds")
```

Comment:
nit: typo "could not determin**ed** the ObjectID..." -> "could not determin**e** the ObjectID".

Also,  by virtue of line 95, the user cannot override the object ID, so he's basically stuck here until we review the next PR...? Maybe do a conditional assignment?
``` go
if b.config.ObjectID == "" {
    b.config.ObjectID = getObjectIdFromToken(spnCloud)
} else {
    // warn the user that he has overridden the object id ?
}
```

@boumenot: is this overkill?

Reply:
Done. I dont override it if its provided but, do let the user know.